src/three/main.ts

```diff
@@ -8,8 +8,12 @@
 /// <reference path="hud.ts" />
 
 module BP3D.Three {
-  export var Main = function (model, element, canvasElement, opts) {
+
+    export var CmToWorld = 1;
+
+    export var Main = function (model, element, canvasElement, opts, alreadyRenderer, alreadyScene: BP3D.Model.Scene) {
```

Comment:
Suggestion here.

Have the BP3D.Three constructor always take a scene and renderer object. Have a factory method to create a DefaultThree object that constructs a default scene and renderer object.

This should be able to avoid any if/else logic within the BP3D.Three object to determine which scene or renderer to use.